src/models/model_loader.cpp

```diff
@@ -134,15 +289,33 @@ bool ModelLoader::exists(const std::string& type) const
 
 // ----------------------------------------------------------------------------------------------------
 
-bool ModelLoader::create(const UUID& id, const std::string& type, UpdateRequest& req, std::stringstream& error)
+bool ModelLoader::create(const UUID& id, const std::string& type, UpdateRequest& req, std::stringstream& error,
+                         const bool allow_sdf)
 {
+    tue::config::DataConstPointer data;
     std::vector<std::string> types;
-    tue::config::DataConstPointer data = loadModelData(type, types, error);
-    if (data.empty())
-        return false;
+    bool sdf = true;
+    if (allow_sdf)
+        data = loadSDFData("model://" + type, error);
+    if(data.empty())
+    {
+        sdf = false;
+        data = loadModelData(type, types, error);
+        if (data.empty())
+            return false;
+    }
 
-    if (!create(data, id, "", req, error))
-        return false;
+    // First to get data before trying to create models
+    if (sdf)
+    {
+        if (!createSDF(data, "", geo::Pose3D::identity(), id, boost::shared_ptr<const geo::Pose3D>(), req, error))
+            return false;
+    }
+    else
+    {
+        if (!create(data, id, "", req, error))
```

Comment:
Quite inconsistent, `create` (without saying what you create) versus `createSDF`.

Reply:
Getting name problems typically means you're trying to push too much different functionality into a single name. Perhaps is also related to comment by Janno about names being too generic.

Fair enough for now though, let's leave this for tomorrow.